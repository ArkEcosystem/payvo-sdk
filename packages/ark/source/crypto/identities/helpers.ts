import { configManager } from "../managers";

export const getWifFromNetwork = (): number => configManager.get("network.wif");
export const getPubKeyHashFromNetwork = (): number => configManager.get("network.pubKeyHash");

export const getPubKeyHash = (): number => configManager.get("network.pubKeyHash");
