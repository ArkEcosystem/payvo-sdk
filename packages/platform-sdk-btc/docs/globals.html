<!doctype html>
<html class="default no-js">
<head>
	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<title>@arkecosystem/platform-sdk-btc</title>
	<meta name="description" content="Documentation for @arkecosystem/platform-sdk-btc">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" href="assets/css/main.css">
</head>
<body>
<header>
	<div class="tsd-page-toolbar">
		<div class="container">
			<div class="table-wrap">
				<div class="table-cell" id="tsd-search" data-index="assets/js/search.json" data-base=".">
					<div class="field">
						<label for="tsd-search-field" class="tsd-widget search no-caption">Search</label>
						<input id="tsd-search-field" type="text" />
					</div>
					<ul class="results">
						<li class="state loading">Preparing search index...</li>
						<li class="state failure">The search index is not available</li>
					</ul>
					<a href="index.html" class="title">@arkecosystem/platform-sdk-btc</a>
				</div>
				<div class="table-cell" id="tsd-widgets">
					<div id="tsd-filter">
						<a href="#" class="tsd-widget options no-caption" data-toggle="options">Options</a>
						<div class="tsd-filter-group">
							<div class="tsd-select" id="tsd-filter-visibility">
								<span class="tsd-select-label">All</span>
								<ul class="tsd-select-list">
									<li data-value="public">Public</li>
									<li data-value="protected">Public/Protected</li>
									<li data-value="private" class="selected">All</li>
								</ul>
							</div>
							<input type="checkbox" id="tsd-filter-inherited" checked />
							<label class="tsd-widget" for="tsd-filter-inherited">Inherited</label>
							<input type="checkbox" id="tsd-filter-externals" checked />
							<label class="tsd-widget" for="tsd-filter-externals">Externals</label>
							<input type="checkbox" id="tsd-filter-only-exported" />
							<label class="tsd-widget" for="tsd-filter-only-exported">Only exported</label>
						</div>
					</div>
					<a href="#" class="tsd-widget menu no-caption" data-toggle="menu">Menu</a>
				</div>
			</div>
		</div>
	</div>
	<div class="tsd-page-title">
		<div class="container">
			<ul class="tsd-breadcrumb">
				<li>
					<a href="globals.html">Globals</a>
				</li>
			</ul>
			<h1>@arkecosystem/platform-sdk-btc</h1>
		</div>
	</div>
</header>
<div class="container container-main">
	<div class="row">
		<div class="col-8 col-content">
			<section class="tsd-panel-group tsd-index-group">
				<h2>Index</h2>
				<section class="tsd-panel tsd-index-panel">
					<div class="tsd-index-content">
						<section class="tsd-index-section ">
							<h3>Modules</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-module"><a href="modules/_src_contracts_.html" class="tsd-kind-icon">&quot;src/contracts&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_index_.html" class="tsd-kind-icon">&quot;src/dto/index&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_peer_.html" class="tsd-kind-icon">&quot;src/dto/peer&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_signed_transaction_.html" class="tsd-kind-icon">&quot;src/dto/signed-<wbr>transaction&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_transaction_.html" class="tsd-kind-icon">&quot;src/dto/transaction&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_transaction_test_.html" class="tsd-kind-icon">&quot;src/dto/transaction.test&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_transactions_bridgechain_registration_.html" class="tsd-kind-icon">&quot;src/dto/transactions/bridgechain-<wbr>registration&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_transactions_bridgechain_resignation_.html" class="tsd-kind-icon">&quot;src/dto/transactions/bridgechain-<wbr>resignation&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_transactions_bridgechain_update_.html" class="tsd-kind-icon">&quot;src/dto/transactions/bridgechain-<wbr>update&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_transactions_business_registration_.html" class="tsd-kind-icon">&quot;src/dto/transactions/business-<wbr>registration&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_transactions_business_resignation_.html" class="tsd-kind-icon">&quot;src/dto/transactions/business-<wbr>resignation&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_transactions_business_update_.html" class="tsd-kind-icon">&quot;src/dto/transactions/business-<wbr>update&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_transactions_delegate_registration_.html" class="tsd-kind-icon">&quot;src/dto/transactions/delegate-<wbr>registration&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_transactions_delegate_resignation_.html" class="tsd-kind-icon">&quot;src/dto/transactions/delegate-<wbr>resignation&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_transactions_entity_registration_.html" class="tsd-kind-icon">&quot;src/dto/transactions/entity-<wbr>registration&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_transactions_entity_resignation_.html" class="tsd-kind-icon">&quot;src/dto/transactions/entity-<wbr>resignation&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_transactions_entity_update_.html" class="tsd-kind-icon">&quot;src/dto/transactions/entity-<wbr>update&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_transactions_htlc_claim_.html" class="tsd-kind-icon">&quot;src/dto/transactions/htlc-<wbr>claim&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_transactions_htlc_lock_.html" class="tsd-kind-icon">&quot;src/dto/transactions/htlc-<wbr>lock&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_transactions_htlc_refund_.html" class="tsd-kind-icon">&quot;src/dto/transactions/htlc-<wbr>refund&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_transactions_index_.html" class="tsd-kind-icon">&quot;src/dto/transactions/index&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_transactions_ipfs_.html" class="tsd-kind-icon">&quot;src/dto/transactions/ipfs&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_transactions_multi_payment_.html" class="tsd-kind-icon">&quot;src/dto/transactions/multi-<wbr>payment&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_transactions_multi_signature_.html" class="tsd-kind-icon">&quot;src/dto/transactions/multi-<wbr>signature&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_transactions_second_signature_.html" class="tsd-kind-icon">&quot;src/dto/transactions/second-<wbr>signature&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_transactions_transfer_.html" class="tsd-kind-icon">&quot;src/dto/transactions/transfer&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_transactions_vote_.html" class="tsd-kind-icon">&quot;src/dto/transactions/vote&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_wallet_.html" class="tsd-kind-icon">&quot;src/dto/wallet&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_dto_wallet_test_.html" class="tsd-kind-icon">&quot;src/dto/wallet.test&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_index_.html" class="tsd-kind-icon">&quot;src/index&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_manifest_.html" class="tsd-kind-icon">&quot;src/manifest&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_schema_.html" class="tsd-kind-icon">&quot;src/schema&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_services_client_.html" class="tsd-kind-icon">&quot;src/services/client&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_services_client_test_.html" class="tsd-kind-icon">&quot;src/services/client.test&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_services_fee_.html" class="tsd-kind-icon">&quot;src/services/fee&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_services_identity_test_.html" class="tsd-kind-icon">&quot;src/services/identity.test&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_services_identity_address_.html" class="tsd-kind-icon">&quot;src/services/identity/address&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_services_identity_index_.html" class="tsd-kind-icon">&quot;src/services/identity/index&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_services_identity_keys_.html" class="tsd-kind-icon">&quot;src/services/identity/keys&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_services_identity_private_key_.html" class="tsd-kind-icon">&quot;src/services/identity/private-<wbr>key&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_services_identity_public_key_.html" class="tsd-kind-icon">&quot;src/services/identity/public-<wbr>key&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_services_identity_utils_.html" class="tsd-kind-icon">&quot;src/services/identity/utils&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_services_identity_wif_.html" class="tsd-kind-icon">&quot;src/services/identity/wif&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_services_index_.html" class="tsd-kind-icon">&quot;src/services/index&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_services_ledger_.html" class="tsd-kind-icon">&quot;src/services/ledger&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_services_ledger_test_.html" class="tsd-kind-icon">&quot;src/services/ledger.test&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_services_link_.html" class="tsd-kind-icon">&quot;src/services/link&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_services_message_.html" class="tsd-kind-icon">&quot;src/services/message&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_services_message_test_.html" class="tsd-kind-icon">&quot;src/services/message.test&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_services_multi_signature_.html" class="tsd-kind-icon">&quot;src/services/multi-<wbr>signature&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_services_peer_.html" class="tsd-kind-icon">&quot;src/services/peer&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_services_transaction_.html" class="tsd-kind-icon">&quot;src/services/transaction&quot;</a></li>
								<li class="tsd-kind-module"><a href="modules/_src_utils_unspent_aggregator_.html" class="tsd-kind-icon">&quot;src/utils/unspent-<wbr>aggregator&quot;</a></li>
								<li class="tsd-kind-module tsd-is-external"><a href="modules/_test_fixtures_client_transaction_.html" class="tsd-kind-icon">&quot;test/fixtures/client/transaction&quot;</a></li>
								<li class="tsd-kind-module tsd-is-external"><a href="modules/_test_fixtures_client_wallet_.html" class="tsd-kind-icon">&quot;test/fixtures/client/wallet&quot;</a></li>
								<li class="tsd-kind-module tsd-is-external"><a href="modules/_test_fixtures_identity_.html" class="tsd-kind-icon">&quot;test/fixtures/identity&quot;</a></li>
								<li class="tsd-kind-module tsd-is-external"><a href="modules/_test_fixtures_ledger_.html" class="tsd-kind-icon">&quot;test/fixtures/ledger&quot;</a></li>
								<li class="tsd-kind-module tsd-is-external"><a href="modules/_test_helpers_.html" class="tsd-kind-icon">&quot;test/helpers&quot;</a></li>
							</ul>
						</section>
					</div>
				</section>
			</section>
		</div>
		<div class="col-4 col-menu menu-sticky-wrap menu-highlight">
			<nav class="tsd-navigation primary">
				<ul>
					<li class="globals current ">
						<a href="globals.html"><em>Globals</em></a>
					</li>
					<li class="label tsd-is-external">
						<span>Internals</span>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_contracts_.html">&quot;src/contracts&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_index_.html">&quot;src/dto/index&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_peer_.html">&quot;src/dto/peer&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_signed_transaction_.html">&quot;src/dto/signed-<wbr>transaction&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_transaction_.html">&quot;src/dto/transaction&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_transaction_test_.html">&quot;src/dto/transaction.test&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_transactions_bridgechain_registration_.html">&quot;src/dto/transactions/bridgechain-<wbr>registration&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_transactions_bridgechain_resignation_.html">&quot;src/dto/transactions/bridgechain-<wbr>resignation&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_transactions_bridgechain_update_.html">&quot;src/dto/transactions/bridgechain-<wbr>update&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_transactions_business_registration_.html">&quot;src/dto/transactions/business-<wbr>registration&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_transactions_business_resignation_.html">&quot;src/dto/transactions/business-<wbr>resignation&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_transactions_business_update_.html">&quot;src/dto/transactions/business-<wbr>update&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_transactions_delegate_registration_.html">&quot;src/dto/transactions/delegate-<wbr>registration&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_transactions_delegate_resignation_.html">&quot;src/dto/transactions/delegate-<wbr>resignation&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_transactions_entity_registration_.html">&quot;src/dto/transactions/entity-<wbr>registration&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_transactions_entity_resignation_.html">&quot;src/dto/transactions/entity-<wbr>resignation&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_transactions_entity_update_.html">&quot;src/dto/transactions/entity-<wbr>update&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_transactions_htlc_claim_.html">&quot;src/dto/transactions/htlc-<wbr>claim&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_transactions_htlc_lock_.html">&quot;src/dto/transactions/htlc-<wbr>lock&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_transactions_htlc_refund_.html">&quot;src/dto/transactions/htlc-<wbr>refund&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_transactions_index_.html">&quot;src/dto/transactions/index&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_transactions_ipfs_.html">&quot;src/dto/transactions/ipfs&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_transactions_multi_payment_.html">&quot;src/dto/transactions/multi-<wbr>payment&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_transactions_multi_signature_.html">&quot;src/dto/transactions/multi-<wbr>signature&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_transactions_second_signature_.html">&quot;src/dto/transactions/second-<wbr>signature&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_transactions_transfer_.html">&quot;src/dto/transactions/transfer&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_transactions_vote_.html">&quot;src/dto/transactions/vote&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_wallet_.html">&quot;src/dto/wallet&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_dto_wallet_test_.html">&quot;src/dto/wallet.test&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_index_.html">&quot;src/index&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_manifest_.html">&quot;src/manifest&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_schema_.html">&quot;src/schema&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_services_client_.html">&quot;src/services/client&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_services_client_test_.html">&quot;src/services/client.test&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_services_fee_.html">&quot;src/services/fee&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_services_identity_test_.html">&quot;src/services/identity.test&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_services_identity_address_.html">&quot;src/services/identity/address&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_services_identity_index_.html">&quot;src/services/identity/index&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_services_identity_keys_.html">&quot;src/services/identity/keys&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_services_identity_private_key_.html">&quot;src/services/identity/private-<wbr>key&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_services_identity_public_key_.html">&quot;src/services/identity/public-<wbr>key&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_services_identity_utils_.html">&quot;src/services/identity/utils&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_services_identity_wif_.html">&quot;src/services/identity/wif&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_services_index_.html">&quot;src/services/index&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_services_ledger_.html">&quot;src/services/ledger&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_services_ledger_test_.html">&quot;src/services/ledger.test&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_services_link_.html">&quot;src/services/link&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_services_message_.html">&quot;src/services/message&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_services_message_test_.html">&quot;src/services/message.test&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_services_multi_signature_.html">&quot;src/services/multi-<wbr>signature&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_services_peer_.html">&quot;src/services/peer&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_services_transaction_.html">&quot;src/services/transaction&quot;</a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/_src_utils_unspent_aggregator_.html">&quot;src/utils/unspent-<wbr>aggregator&quot;</a>
					</li>
					<li class="label tsd-is-external">
						<span>Externals</span>
					</li>
					<li class=" tsd-kind-module tsd-is-external">
						<a href="modules/_test_fixtures_client_transaction_.html">&quot;test/fixtures/client/transaction&quot;</a>
					</li>
					<li class=" tsd-kind-module tsd-is-external">
						<a href="modules/_test_fixtures_client_wallet_.html">&quot;test/fixtures/client/wallet&quot;</a>
					</li>
					<li class=" tsd-kind-module tsd-is-external">
						<a href="modules/_test_fixtures_identity_.html">&quot;test/fixtures/identity&quot;</a>
					</li>
					<li class=" tsd-kind-module tsd-is-external">
						<a href="modules/_test_fixtures_ledger_.html">&quot;test/fixtures/ledger&quot;</a>
					</li>
					<li class=" tsd-kind-module tsd-is-external">
						<a href="modules/_test_helpers_.html">&quot;test/helpers&quot;</a>
					</li>
				</ul>
			</nav>
			<nav class="tsd-navigation secondary menu-sticky">
				<ul class="before-current">
				</ul>
			</nav>
		</div>
	</div>
</div>
<footer class="with-border-bottom">
	<div class="container">
		<h2>Legend</h2>
		<div class="tsd-legend-group">
			<ul class="tsd-legend">
				<li class="tsd-kind-object-literal"><span class="tsd-kind-icon">Object literal</span></li>
				<li class="tsd-kind-variable"><span class="tsd-kind-icon">Variable</span></li>
				<li class="tsd-kind-function"><span class="tsd-kind-icon">Function</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-interface"><span class="tsd-kind-icon">Interface</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-class"><span class="tsd-kind-icon">Class</span></li>
			</ul>
		</div>
	</div>
</footer>
<div class="container tsd-generator">
	<p>Generated using <a href="https://typedoc.org/" target="_blank">TypeDoc</a></p>
</div>
<div class="overlay"></div>
<script src="assets/js/main.js"></script>
</body>
</html>