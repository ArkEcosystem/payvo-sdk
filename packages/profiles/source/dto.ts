export * from "./signed-transaction.dto.js";
export * from "./transaction.collection";
export * from "./transaction.dto.js";
export * from "./transaction.mapper";
