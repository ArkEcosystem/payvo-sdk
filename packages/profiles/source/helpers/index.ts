export * from "./avatar";
export * from "./currency";
export * from "./promise";
export * from "./queue";
