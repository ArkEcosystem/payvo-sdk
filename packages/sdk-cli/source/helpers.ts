import { ADA } from "@payvo/sdk-ada";
import { ARK } from "@payvo/sdk-ark";
// import { ATOM } from "@payvo/sdk-atom";
import { AVAX } from "@payvo/sdk-avax";
// import { BTC } from "@payvo/sdk-btc";
import { DOT } from "@payvo/sdk-dot";
import { EGLD } from "@payvo/sdk-egld";
import { TRX } from "@payvo/sdk-trx";
import { XLM } from "@payvo/sdk-xlm";
// import { XRP } from "@payvo/sdk-xrp";
import { Request } from "@payvo/sdk-http-got";
// import { EOS } from "@payvo/sdk-eos";
// import { ETH } from "@payvo/sdk-eth";
import { LSK } from "@payvo/sdk-lsk";
import { Environment } from "@payvo/sdk-profiles";
// import { NEO } from "@payvo/sdk-neo";
import { SOL } from "@payvo/sdk-sol";
import cfonts from "cfonts";

import { ConfStorage } from "./storage";

export const useLogger = (): Console => console;

export const useEnvironment = async (): Promise<Environment> => {
	const env = new Environment({
		coins: {
			ADA,
			ARK,
			// ATOM,
			AVAX,
			// BTC,
			DOT,
			EGLD,
			// EOS,
			// ETH,
			LSK,
			// NEO,
			SOL,
			TRX,
			XLM,
			// XRP,
		},
		storage: new ConfStorage(),
		httpClient: new Request(),
	});

	await env.verify();
	await env.boot();

	return env;
};

export const renderLogo = (): void => {
	console.clear();

	cfonts.say("Plutus", {
		gradient: ["#14b8a6", "#0891b2"],
		independentGradient: true,
		transitionGradient: true,
	});
};
