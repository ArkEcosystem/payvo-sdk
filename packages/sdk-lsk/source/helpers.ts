export const calculateBalanceLockedInVotes = (votes = {}) =>
	Object.values(votes).reduce((total, vote: any) => total + vote.confirmed, 0);

export const calculateBalanceLockedInUnvotes = (unlocking = []) =>
	unlocking.reduce((acc, vote: any) => acc + parseInt(vote.amount, 10), 0);

export const isBlockHeightReached = (unlockHeight, currentBlockHeight) => currentBlockHeight >= unlockHeight;

export const getUnlockableUnlockObjects = (unlocking = [], currentBlockHeight = 0) =>
	unlocking
		.filter((vote: any) => isBlockHeightReached(vote.height.end, currentBlockHeight))
		.map((vote: any) => ({
			delegateAddress: vote.delegateAddress,
			amount: vote.amount,
			unvoteHeight: Number(vote.height.start),
		}));

export const calculateUnlockableBalance = (unlocking = [], currentBlockHeight = 0) =>
	unlocking.reduce(
		(sum, vote: any) =>
			isBlockHeightReached(vote.height.end, currentBlockHeight) ? sum + parseInt(vote.amount, 10) : sum,
		0,
	);

export const calculateBalanceUnlockableInTheFuture = (unlocking = [], currentBlockHeight = 0) =>
	unlocking.reduce(
		(sum, vote: any) =>
			!isBlockHeightReached(vote.height.end, currentBlockHeight) ? sum + parseInt(vote.amount, 10) : sum,
		0,
	);

export const isTransfer = ({ assetID, moduleID }: Record<string, any>): boolean => moduleID === 2 && assetID === 0;

export const isMultiSignatureRegistration = ({ assetID, moduleID }: Record<string, any>): boolean =>
	moduleID === 4 && assetID === 0;

export const isDelegateRegistration = ({ assetID, moduleID }: Record<string, any>): boolean =>
	moduleID === 5 && assetID === 0;

export const isVote = ({ assetID, moduleID }: Record<string, any>): boolean => moduleID === 5 && assetID === 1;
