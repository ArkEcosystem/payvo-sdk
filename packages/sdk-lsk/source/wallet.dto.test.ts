import { BigNumber } from "@payvo/helpers";

import Fixture from "../test/fixtures/client/wallet.json";
import FixtureLegacy0 from "../test/fixtures/client/wallet-legacy-0.json";
import FixtureLegacy1 from "../test/fixtures/client/wallet-legacy-1.json";
import { WalletData } from "./wallet.dto";
import { createService } from "../test/mocking";
import { expect } from "@jest/globals";

let subject: WalletData;

describe("WalletData", () => {
	describe.each([FixtureLegacy0, FixtureLegacy1])("2.0", (fixture) => {
		beforeEach(async () => (subject = (await createService(WalletData)).fill(fixture.data[0])));

		test("#address", async () => {
			expect(subject.address()).toBe("11603034586667438647L");
		});

		test("#publicKey", () => {
			expect(subject.publicKey()).toBe("635ec78f25eefe8dcb7b4e7f5685c5aad1fef54048c596ee4e74cdf5e7b0f37c");
		});

		test("#nonce", () => {
			expect(subject.nonce()).toBeInstanceOf(BigNumber);
			expect(subject.nonce().toHuman()).toBe(0);
		});

		test("#username", () => {
			expect(subject.username()).toBe("username");
		});

		test("#rank", () => {
			expect(subject.rank()).toBe(1);
		});

		test("#vote", () => {
			expect(subject.votes()).toBeInstanceOf(BigNumber);
			expect(subject.votes()?.toHuman()).toBe(1);
		});

		test("#multiSignature", () => {
			expect(subject.multiSignature()).toMatchSnapshot();
		});
	});

	describe("3.0", () => {
		beforeEach(async () => (subject = (await createService(WalletData)).fill(Fixture.data[0])));

		test("#address", () => {
			expect(subject.address()).toBe("lskk8upba9sj8zsktr8hb2vcgk3quvgmx8h27h4gr");
		});

		test("#publicKey", () => {
			expect(subject.publicKey()).toBe("414934d5c70dec65c4c01ddef4cb131913cc53b18e0c1c375857a5e7db52484b");
		});

		test("#balance", () => {
			expect(subject.balance().total).toBeInstanceOf(BigNumber);
			expect(subject.balance().total.toHuman()).toBe(1509.94716);

			expect(subject.balance().available).toBeInstanceOf(BigNumber);
			expect(subject.balance().available.toHuman()).toBe(1489.94716);

			expect(subject.balance().fees).toBeInstanceOf(BigNumber);
			expect(subject.balance().fees.toHuman()).toBe(1489.94716);

			expect(subject.balance().locked).toBeInstanceOf(BigNumber);
			expect(subject.balance().locked?.toHuman()).toBe(20);

			expect(subject.balance().lockedVotes).toBeInstanceOf(BigNumber);
			expect(subject.balance().lockedVotes?.toHuman()).toBe(10);

			expect(subject.balance().lockedUnvotes).toBeInstanceOf(BigNumber);
			expect(subject.balance().lockedUnvotes?.toHuman()).toBe(10);
		});

		test("#isDelegate", () => {
			expect(subject.isDelegate()).toBe(true);
		});

		describe("#multiSignature", () => {
			it("should throw error if wallet has not registered multi-signature", () => {
				expect(() => subject.multiSignature()).toThrowError(
					"This wallet does not have a multi-signature registered.",
				);
			});
		});

		describe("#votes", () => {
			it("should return number of votes received", () => {
				expect(subject.votes().toHuman()).toBe(0);
			});

			it("should default to 0", async () => {
				const votes = (await createService(WalletData)).fill({}).votes().toHuman();

				expect(votes).toBe(0);
			});
		});
	});
});
