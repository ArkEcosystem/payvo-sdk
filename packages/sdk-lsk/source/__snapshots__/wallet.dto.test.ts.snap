// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletData 2.0 #multiSignature 1`] = `
Object {
  "mandatoryKeys": Array [],
  "numberOfSignatures": 0,
  "optionalKeys": Array [],
}
`;

exports[`WalletData 2.0 #multiSignature 2`] = `
Object {
  "mandatoryKeys": Array [],
  "numberOfSignatures": 0,
  "optionalKeys": Array [],
}
`;
